### PR TITLE
support/feature/chat-screen

### PR DESCRIPTION
**Pull request by devsadeq**
- Create support chat screen ui and handle it with fake data.
- Create Idle state
- Make BpSimpleTextField take icons interactions listener.

java_eRpxUbCAkR

### CODE REVIEW

File: client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/chat/ChatScreen.kt [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                    .padding(end = if (message.isMe) 0.kms else 400.kms)
+                    .clip(RoundedCornerShape(Theme.radius.medium))
+                    .background(if (message.isMe) Theme.colors.background else Theme.colors.secondary)
+                    .padding(vertical = 16.kms, horizontal = 24.kms)
+            )
+        }
+    }
+
+    @Composable
+    private fun Idle() {
```
Comment (AzizaHelmy) [inline client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/chat/ChatScreen.kt:172]: Method name should be more descriptive

**Review by AzizaHelmy — APPROVED** [review]

**Review by hassanwasfy — APPROVED** [review]
Well done ✅

File: design_system/shared/src/commonMain/kotlin/ui/composable/BpSimpleTextField.kt [hunk 2]
```diff
@@ -38,62 +34,69 @@ fun BpSimpleTextField(
...
         )
         AnimatedVisibility(isError) {
             Text(
-                    text = errorMessage,
-                    modifier = Modifier.padding(top = Theme.dimens.space8),
-                    style = Theme.typography.caption,
-                    color = Theme.colors.primary
+                text = errorMessage,
+                modifier = Modifier.padding(top = Theme.dimens.space8),
+                style = Theme.typography.caption,
```
Comment (ay0o5h) [inline design_system/shared/src/commonMain/kotlin/ui/composable/BpSimpleTextField.kt:98]: No need to use space8, from theme just write 8.dp

File: client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/composable/modifier/CircleLayout.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.layout
+
+@Stable
+fun Modifier.circleLayout(color: Color = Color(0xffF53D47)): Modifier {
```
Comment (ay0o5h) [inline client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/composable/modifier/CircleLayout.kt:11]: Use color from design system

**Review by ay0o5h — CHANGES_REQUESTED** [review]

**Review by ay0o5h — APPROVED** [review]

File: client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/chat/ChatScreen.kt [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+                .background(Theme.colors.surface)
+        ) {
+            MainAppbar(
+                username = state.appbar.username,
+                isDropMenuExpanded = state.appbar.isDropdownMenuExpanded,
+                onClickDropDownMenu = listener::onClickDropdownMenu,
+                onDismissDropDownMenu = listener::onDismissDropdownMenu,
+                onLogOut = listener::onClickLogOut,
+            )
+            if (state.idle) IdlePlaceholder() else ChatScreenContent(state, listener)
```
Comment (mosayed01) [inline client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/chat/ChatScreen.kt:52]:
```suggestion
            AnimatedVisibilty (state.idle) IdlePlaceholder() else ChatScreenContent(state, listener)
```

Comment (mosayed01) [reply client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/chat/ChatScreen.kt]: Sorry I mean AnimatedContent

Comment (mosayed01) [reply client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/chat/ChatScreen.kt]: @decsadeq

File: client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/chat/ChatScreenModel.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+            it.copy(message = message)
+        }
+    }
+
+    override fun onSendMessageClicked() {
+        tryToExecute(
+            {
+                manageChatUseCase.sendMessage(
+                    NewMessage(
+                        ticketId = state.value.ticket.id,
```
Comment (Asiasama710) [inline client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/chat/ChatScreenModel.kt:83]: i think u should use `mutableState ` not  `state`

Comment (devsadeq) [reply client_support/common/src/commonMain/kotlin/org/thechance/common/presentation/chat/ChatScreenModel.kt:83]: Why do I need to use mutable state? I only need to retrieve the value without changing it.

**Review by mosayed01 — APPROVED** [review]
LGTM

**Review by aya-seif — APPROVED** [review]

**Pull request merged**